sql/core/src/main/scala/org/apache/spark/sql/execution/command/AnalyzeTableCommand.scala

```diff
@@ -88,14 +89,70 @@ case class AnalyzeTableCommand(tableName: String) extends RunnableCommand {
             }
           }.getOrElse(0L)
 
-        // Update the Hive metastore if the total size of the table is different than the size
-        // recorded in the Hive metastore.
-        // This logic is based on org.apache.hadoop.hive.ql.exec.StatsTask.aggregateStats().
-        if (newTotalSize > 0 && newTotalSize != oldTotalSize) {
+        var needUpdate = false
+        val totalSize = if (newTotalSize > 0 && newTotalSize != oldTotalSize) {
+          needUpdate = true
+          newTotalSize
+        } else {
+          oldTotalSize
+        }
+        var numRows: Option[BigInt] = None
+        if (!noscan) {
+          val oldRowCount: Long = if (catalogTable.catalogStats.isDefined) {
+            catalogTable.catalogStats.get.rowCount.map(_.toLong).getOrElse(-1L)
+          } else {
+            -1L
+          }
+          val newRowCount = sparkSession.table(tableName).count()
+          if (newRowCount >= 0 && newRowCount != oldRowCount) {
```

Comment:
If we delete the statistics, we can't tell whether we don't collect stats or the table is empty.